web/base_viewer.js

```diff
@@ -1475,6 +1481,7 @@ class BaseViewer {
     }
     this._setCurrentPageNumber(pageNumber, /* resetCurrentPageView = */ true);
     this.update();
+    this._setScale(this._currentScaleValue, true);
```

Comment:
Please do exactly that same as in https://github.com/mozilla/pdf.js/blob/d784af3f38d5604b653c06fb40a6fe6a021d45d3/web/base_viewer.js#L1412-L1416 here as well, since this call shouldn't be needed unconditionally!